src/main/java/com/google/firebase/FirebaseApp.java

```diff
@@ -427,28 +338,49 @@ FirebaseService getService(String id) {
    * its own instance of this class. This class is not thread safe. The caller (FirebaseApp) must
    * ensure that methods are called serially.
    */
-  static class TokenRefresher {
+  static class TokenRefresher implements CredentialsChangedListener {
```

Comment:
It seems like you've reworked the way this works significantly (we schedule refreshes which trigger changes which cause us to schedule refreshes). I think it's probably worth revisiting the class comment to make sure it explains the intended approach.

Reply:
Good catch. Updated the class-level javadoc.